doc/lua_api.txt

```diff
@@ -5470,8 +5470,14 @@ This is basically a reference to a C++ `ServerActiveObject`
     * `acc` is a vector
 * `get_acceleration()`: returns the acceleration, a vector
 * `set_rotation(rot)`
-    * `rot` is a vector (radians). X is pitch (elevation), Y is yaw (heading)
-      and Z is roll (bank).
+    * `rot` is a vector (radians).
+    * `rot.y` is rotation around the world Y axis, and acts like `set_yaw()`.
+    * `rot.x` is rotation around the object's left-right axis, so can be considered
+      to be 'pitch'. The object's left-right axis is rotated around the world Y axis
+      by rot.y (yaw).
+    * `rot.z is rotation around the object's front-back axis, so can be considered
+      to be 'roll'. The object's front-back axis is rotated around the object's
+      left-right axis by rot.x (pitch).
```

Comment:
The object is first rotated around the z-axis by `rot.z` (roll), then around the x-axis by `rot.x` (pitch) and lastly around the y-axis by `rot.y` (yaw). All these rotations are counter-clock-wise, I think.
The "front-back" axis is also rotated by yaw, not only by pitch.
The front can be considered to be in (-?)z direction.

Reply:
Well, you have the correct sequence but backwards, maybe because you are understanding it in a different way.
It's clearer to understand it 'from world to object' in terms of a gimbal system.
> The "front-back" axis is also rotated by yaw, not only by pitch.

Yes, but that's implicit in
`The object's front-back axis is rotated around the object's left-right axis`
because
`The object's left-right axis is rotated around the world Y axis by rot.y'

Reply:
Well I could write: _Your intrinsic sequence is backwards!_ \:P

> > The "front-back" axis is also rotated by yaw, not only by pitch.
> 
> Yes, but that's implicit in
> `The object's front-back axis is rotated around the object's left-right axis`
> because
> `The object's left-right axis is rotated around the world Y axis by rot.y`

No it isn't implicit. Let's take an example:
yaw: 90°, pitch: 45°
(initial front: (0,0,-1), initial left: (1,0,0), ccw rotation)
after yaw: front: (1,0,0), left: (0,0,1)
after pitch (rotation around (0,0,1)): front: (sqrt(2)/2,sqrt(2)/2,0) (= the actual roll axis), rotated front-back axis (= rotated initial front): still (0,0,-1)
TL;DR: Rotation around a rotated axis is not the same as two rotations.

Reply:
That doesn't make sense to me.

Reply:
Woahh ... i just realised a mistake, the object left-right axis rotates when roll is applied, so i can't use that, instead i have to write about the 'pitch axis' which is different.
Your comments helped me think further.
So WIP, will try to improve.